quickstarts/microsoft.datafactory/data-factory-get-started/main.bicep

```diff
@@ -0,0 +1,199 @@
+param location string = resourceGroup().location
+
+@description('Name of the Azure storage account that contains the input/output data.')
+param storageAccountName string = 'storage${uniqueString(resourceGroup().id)}'
+
+@description('Name of the blob container in the Azure Storage account.')
+param blobContainerName string = 'blob${uniqueString(resourceGroup().id)}'
+
+@description('Name of the managed identity created to access the Azure Storage account.')
+param managedIdentityName string = 'blob${uniqueString(resourceGroup().id)}'
+
+var dataFactoryLinkedServiceName = 'ArmtemplateStorageLinkedService'
+var dataFactoryDataSetInName = 'ArmtemplateTestDatasetIn'
+var dataFactoryDataSetOutName = 'ArmtemplateTestDatasetOut'
+var pipelineName = 'ArmtemplateSampleCopyPipeline'
+var bootstrapRoleAssignmentId_var = guid('${resourceGroup().id}contributor')
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2021-09-01' = {
+  name: storageAccountName
+  location: location
+  sku: {
+    name: 'Standard_LRS'
+  }
+  kind: 'StorageV2'
+}
+
+resource blobContainer 'Microsoft.Storage/storageAccounts/blobServices/containers@2021-09-01' = {
+  name: '${storageAccountName}/default/${blobContainerName}'
+  dependsOn: [
+    storageAccount
+  ]
+}
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2022-01-31-preview' = {
+  name: managedIdentityName
+  location: location
+}
+
+resource bootstrapRoleAssignmentId 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: bootstrapRoleAssignmentId_var
+  properties: {
+    roleDefinitionId: '/subscriptions/${subscription().subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c'
+    principalId: reference(managedIdentity.id, '2022-01-31-preview').principalId
+    principalType: 'ServicePrincipal'
+  }
+}
+
+resource runPowerShellInlineWithOutput 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
+  name: 'copyFile'
+  location: location
+  kind: 'AzurePowerShell'
+  identity: {
+    type: 'UserAssigned'
+    userAssignedIdentities: {
+      '${managedIdentity.id}': {
+      }
+    }
+  }
+  properties: {
+    azPowerShellVersion: '6.4'
+    arguments: '-name FileCreationScript'
+    environmentVariables: [
+      {
+        name: 'RGName'
+        value: resourceGroup().name
+      }
+      {
+        name: 'SAName'
+        value: storageAccountName
+      }
+      {
+        name: 'ContainerName'
+        value: blobContainerName
+      }
+    ]
+    scriptContent: 'Invoke-WebRequest -Uri \'https://raw.githubusercontent.com/kromerm/adfdataflowdocs/master/sampledata/moviesDB2.csv\' -OutFile \'moviesDB2.csv\'; $storageAccount = Get-AzStorageAccount -ResourceGroupName \${Env:RGName} -Name \${Env:SAName}; $ctx = $storageAccount.Context; Set-AzStorageBlobContent -Container \${Env:ContainerName} -Blob \'input/moviesDB2.csv\' -Context $ctx -StandardBlobTier \'Hot\' -File \'moviesDB2.csv\''
```

Comment:
you can't pull content from a private repo - you can add the csv file to this sample or we can stage it